### PR TITLE
Add races to tc.state

### PR DESCRIPTION
**Pull request by syhw**
adding races to state via handshake

### CODE REVIEW

File: examples/simple_dll.lua [hunk 1]
```diff
@@ -77,6 +77,7 @@ while total_battles < 40 do
     if DEBUG > 1 then
         print('Received init: ', update)
     end
+    print("I'm race: ", tc.state.player_races[tc.state.player_id])
```
Comment (ebetica) [inline examples/simple_dll.lua]: Also print your opponent's race?

**Comment by ebetica** [comment]
Is this saved in the replayer somewhere?

**Comment by syhw** [comment]
Nope, not saved in the replayer, I didn't think it's necessary, do you?

**Comment by syhw** [comment]
See examples in:
* C++ https://github.com/TorchCraft/TorchCraft/pull/192/files#diff-a1f4e8194fa56a8f21ed78a2ae3a5bde
* Python https://github.com/TorchCraft/TorchCraft/pull/192/files#diff-1699d7fdfdbef4851f25ae2093fe31ea
* Lua https://github.com/TorchCraft/TorchCraft/pull/192/files#diff-249f930a45b2c97412b1133784d00475

**Comment by ebetica** [comment]
We can always put it in the replay later if needed

**Pull request merged**